lib/matplotlib/widgets.py

```diff
@@ -1088,7 +1088,7 @@ def __init__(self, ax, labels, actives=None, *, useblit=True,
 
     def _clear(self, event):
         """Internal event handler to clear the buttons."""
-        if self.ignore(event):
+        if self.ignore(event) or self.canvas is not self.ax.figure.canvas:
```

Comment:
I wonder if you want to wrap this in a small method with an explanation like your explanation below as the doc string:





```
def _not_orig_canvas():
    """
    Explain
    """"
    return self.canvas is not self.ax.figure.canvas
```
Perhaps thats too much cruft, but otherwise this is a bit of a mysterious clause.

Reply:
done.